moped-editor/src/views/projects/newProjectView/NewProjectMap.js

```diff
@@ -203,6 +247,81 @@ const NewProjectMap = ({
     saveActionDispatch
   );
 
+  /**
+   * Handles when a new map feature is selected and updates feature collection state
+   * @param {Object} selectedFeature - The selected feature geojson
+   * @return {undefined} Side effect of calling setFeatureCollection with new state
+   */
+  const handleSelectedFeatureUpdate = selectedFeature => {
```

Comment:
this is a lot of code to run on every render—I could hoist it out of the component? `useCallback` might by the right choice?

Reply:
@johnclary we should try useCallback, if the code really gets run every render the gains should be obvious. I am not sure hoisting will improve performance, from what I know running something like useYourComponentHook method would initialize new functions inside of it every run and would not have the referential equality performance that useCallback provides. 

In any case, this is a great opportunity to learn and exercise some react profiling. 🔍 📈